optuna/integration/sklearn.py

```diff
@@ -48,10 +48,10 @@
 if not _imports.is_successful():
     BaseEstimator = object  # NOQA
 
-ArrayLikeType = Union[List, np.ndarray, "pd.Series", spmatrix]
+ArrayLikeType = Union[List, np.ndarray, "pd.Series", "spmatrix"]
 OneDimArrayLikeType = Union[List[float], np.ndarray, "pd.Series"]
-TwoDimArrayLikeType = Union[List[List[float]], np.ndarray, "pd.DataFrame", spmatrix]
-IterableType = Union[List, "pd.DataFrame", np.ndarray, "pd.Series", spmatrix, None]
+TwoDimArrayLikeType = Union[List[List[float]], np.ndarray, "pd.DataFrame", "spmatrix"]
+IterableType = Union[List, "pd.DataFrame", np.ndarray, "pd.Series", "spmatrix", None]
 IndexableType = Union[Iterable, None]
```

Comment:
This suggestion isn't directly related to this PR, but how about using `Optional`.
```suggestion
IndexableType = Optional[Iterable]
```